tortoise/tortoise.go

```diff
@@ -680,22 +676,21 @@ func (t *turtle) onBallot(ballot *types.Ballot) error {
 			t.logger.With().Warning("ref ballot not in state",
 				log.Stringer("ref", ballot.RefBallot),
 			)
-			return nil
+			return nil, nil
 		}
 		if ref.reference == nil {
-			t.logger.With().Warning("invalid ballot used as a reference",
-				log.Stringer("ref", ballot.RefBallot),
-			)
-			return nil
+			return nil, fmt.Errorf("invalid ballot use as a reference %s", ballot.RefBallot)
 		}
 		refinfo = ref.reference
 	}
+
 	if !ballot.IsMalicious() {
```

Comment:
how can this be true now that malicious is set after this method returns.

Reply:
this is true when previously stored ballot is loaded into tortoise